pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -107,16 +131,38 @@ func initDrainer(r *ReconcileNodeMaintenance, config *rest.Config) error {
 	//Label selector to filter pods on the node
 	//r.drainer.PodSelector = "kubevirt.io=virt-launcher"
 
-	cs, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		return err
-	}
+
 	r.drainer.Client = cs
 	r.drainer.DryRun = false
 
+	isLeaseSupported, err := checkLeaseSupported(r.drainer.Client);
+	if err != nil {
+		log.Errorf("Error while checking for lease support %v", err)
```

Comment:
why not return an error here and only log it ?

Reply:
Older systems don't support leases (like os3.11) but the nmo should still work on them.

Reply:
> Older systems don't support leases (like os3.11) but the nmo should still work on them. If checkLeaseSupported returns false that isn't an error in itself 

`checkLeaseSupported()` should not be returning an error in this case
it should only return an error in the case that it could not determine a value for `isLeaseSupported`

> - if the discovery client had an error while checking for leases than that's something we want to know about.

and that error should be propagated to the caller of `initDrainer()`